### PR TITLE
[PF-1742] Show data source resources in `terra resource describe`

### PR DESCRIPTION
**Pull request by melissachang**
```
~/terra-cli (mc/data-source-resource $): terra resource describe --name=1000-genomes
Title:			1000 Genomes Phase 3
Short description:	The genomes of more than a thousand anonymous participants from a number of different ethnic groups were analyzed and made publicly available.
Version:		Phase 3
Resources:
                        NAME                  RESOURCE TYPE         DESCRIPTION
                        gcs-folder            GCS_OBJECT            (unset)
                        pedigree              BQ_TABLE              (unset)
                        phase-3-variants      BQ_TABLE              (unset)
                        sample_info           BQ_TABLE              (unset)
~/terra-cli (mc/data-source-resource $): terra resource describe --name=1000-genomes --format=json
{
  "id" : "62639f59-4cde-487a-b1ca-49b70d6a5d28",
  "name" : "1000-genomes",
  "description" : null,
  "resourceType" : "DATA_SOURCE",
  "stewardshipType" : "REFERENCED",
  "cloningInstructions" : "COPY_NOTHING",
  "accessScope" : null,
  "managedBy" : null,
  "privateUserName" : null,
  "privateUserRoles" : null,
  "dataSourceWorkspaceUuid" : "abdf795a-0066-43a0-aaeb-dd0f1c40526b",
  "title" : "1000 Genomes Phase 3",
  "shortDescription" : "The genomes of more than a thousand anonymous participants from a number of different ethnic groups were analyzed and made publicly available.",
  "version" : "Phase 3",
  "resources" : [ {
    "id" : "b3f8a480-14f5-4583-89f8-7cb8841c4f15",
    "name" : "gcs-folder",
    "description" : null,
    "resourceType" : "GCS_OBJECT",
    "stewardshipType" : "REFERENCED",
    "cloningInstructions" : "COPY_REFERENCE",
    "accessScope" : null,
    "managedBy" : null,
    "privateUserName" : null,
    "privateUserRoles" : null,
    "bucketName" : "genomics-public-data",
    "contentType" : null,
    "objectName" : "1000-genomes-phase-3",
    "isDirectory" : null,
    "size" : null,
    "timeStorageClassUpdated" : null
  }, {
    "id" : "4488d1c8-e544-41d0-abb3-28f17c01cef7",
    "name" : "pedigree",
    "description" : null,
    "resourceType" : "BQ_TABLE",
    "stewardshipType" : "REFERENCED",
    "cloningInstructions" : "COPY_REFERENCE",
    "accessScope" : null,
    "managedBy" : null,
    "privateUserName" : null,
    "privateUserRoles" : null,
    "projectId" : "bigquery-public-data",
    "datasetId" : "human_genome_variants",
    "dataTableId" : "1000_genomes_pedigree",
    "tableDescription" : "Source:\nhttp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_g1k.ped\n\nDescription:\nhttp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/README_20130606_sample_info\n\nFor more information see https://cloud.google.com/genomics/docs/public-datasets/1000-genomes",
    "numRows" : 3501
  }, {
...
  } ]
}
```

### CODE REVIEW

**Review by ginaygoogle — APPROVED** [review]

File: src/main/java/bio/terra/cli/serialization/userfacing/resource/UFDataSource.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+  /** Constructor for Jackson deserialization during testing. */
+  private UFDataSource(Builder builder) {
+    super(builder);
+    this.dataSourceWorkspaceUuid = builder.dataSourceWorkspaceUuid;
+    this.title = builder.title;
+    this.shortDescription = builder.shortDescription;
+    this.version = builder.version;
+    this.resources = builder.resources;
+    ;
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/serialization/userfacing/resource/UFDataSource.java:58]: nit: remove

File: src/main/java/bio/terra/cli/service/WorkspaceManagerService.java [hunk 2]
```diff
@@ -892,6 +897,24 @@ public GcpBigQueryDatasetResource createControlledBigQueryDataset(
         "Error creating controlled BigQuery dataset in the workspace.");
   }
 
+  /** Used by tests only. */
+  public TerraWorkspaceResource createReferencedTerraWorkspace(
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerService.java:901]: I'm still uncomfortable about having an asymetrical commands for datasource. this means before the UI have datasource, this command is useless in the CLI. i'm fine with this going in but i'd like Jenn's opinion on the CLI interface.

**Review by yuhuyoyo — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/resource/DataSource.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  /** Resolve data source. */
+  // TODO(PF-1743, PF-1744): Implement
+  public String resolve() {
+    return "";
+  }
+
+  public Workspace getDataSourceWorkspace() {
+    return Workspace.get(dataSourceWorkspaceUuid);
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/cli/businessobject/resource/DataSource.java:79]: what happens if user tries to describe a datasource that they don't have access to?

**Review by yuhuyoyo — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/resource/DataSource.java (hunk 3 above)
Comment (yuhuyoyo) [reply src/main/java/bio/terra/cli/businessobject/resource/DataSource.java]: thanks! we should have a custom error message so it doesn't expose that it's a workspace. doesn't need to be in this PR though.

**Pull request merged**